src/status_im/contexts/wallet/router/core.cljs

```diff
@@ -0,0 +1,87 @@
+(ns status-im.contexts.wallet.router.core
+  (:require [schema.core :as schema]
+            [status-im.contexts.wallet.router.schema :as router.schema]
+            [status-im.contexts.wallet.send.transaction-settings.core :as transaction-settings]
+            [status-im.contexts.wallet.send.utils :as send-utils]))
+
+(def ^:private gwei-precision 6)
+
+(defn- to-gwei
+  [amount]
+  (send-utils/convert-to-gwei amount gwei-precision))
+
+(defn transaction-fees-by-mode
+  [route]
+  (let [{:keys [low medium high low-estimated-time
+                medium-estimated-time high-estimated-time]}
+        (:suggested-levels-for-max-fees-per-gas route)]
+    {:tx-fee-mode/normal
+     {:max-fees       (to-gwei low)
+      :estimated-time low-estimated-time}
+
+     :tx-fee-mode/fast
+     {:max-fees       (to-gwei medium)
+      :estimated-time medium-estimated-time}
+
+     :tx-fee-mode/urgent
+     {:max-fees       (to-gwei high)
+      :estimated-time high-estimated-time}}))
+
+(schema/=> transaction-fees-by-mode
+  [:=>
+   [:cat router.schema/?route]
+   [:map-of
+    router.schema/?fee-modes
+    [:map {:closed? true}
+     [:max-fees :string]
+     [:estimated-time :int]]]])
+
+(defn transaction-fee-mode
+  [route]
+  (-> route
+      :tx-gas-fee-mode
+      transaction-settings/gas-rate->tx-fee-mode))
+
+(schema/=> transaction-fee-mode
+  [:=>
+   [:cat router.schema/?route]
+   router.schema/?fee-modes])
+
+(defn transaction-estimated-time
```

Comment:
I don't see any unit tests in the PR @clauxx. It's concerning that we are relying exclusively on manual tests from QA and that after so many discussions in our teams we still struggle with something so basic. The problem didn't originate from this PR, I understand you are already doing amazing work with adding the schemas 💯 I would at least cover with unit tests the data store functions because that's our basic contract where data is returned exactly as we want and it's all pure without the shenanigans of re-frame and state management.

I leave it up to you to judge when/how to approach this feedback.

Reply:
Side note: the schema generators could then be used to generate large maps and then assoc/dissoc only as needed. It's an approach I've used/verified in Status (and in a past job) and can get a long way in helping reduce the noise in tests so that we can focus on the important test arrangement and assertions.

Reply:
fully agreed on the tests here and good call on schema generators. Buuut, there was an issue with the regex schema (`:re` is used in the hex schema), which doesn't have a generator in cljs. I added a default return value for the generator, but lmk if you see a better solution.

Reply:
Correct @clauxx there's this extra challenge with generators for regexps. I thought Malli in CLJS had support for generating strings from regexps. I see now in the source there's a generator, but only for Clojure https://github.com/metosin/malli/blob/0.17.0/src/malli/generator.cljc#L193. And the problem was also attacked for clojure.spec https://github.com/miner/strgen

Your solution for works well for now 👍🏼 Thanks!